71 user can change the language support for internationalization

VSirvio opened the pull request:
**Changes:**
- Add `react-i18next` internationalization framework to `frontend/package.json`
- Set up `react-i18next` in `frontend/src/i18n.js`
- Add language selection links to the top right corner of each page
- Add English and Finnish translations to the file `frontend/src/translations.json` for each UI text

amandahamynen reviewed (approved):
Excellent job!

I noticed that when error messages are fetched from the backend, the specific error content (error.response.data) isn’t affected by the translations. I’m sure you’re already aware of this, but I just wanted to mention it. Since there has been some discussion about improving the error messages in general, this might be something to address in the future.

Also, I observed that the language switches back to 'en' after refreshing the page, even if it was previously set to 'fi'. This could be another area to consider for future improvements.

Great work as always, @VSirvio ! I'm happy to approve these changes and will accept the merge once the conflicts are resolved!

VSirvio commented:
> I noticed that when error messages are fetched from the backend, the specific error content (error.response.data) isn’t affected by the translations. I’m sure you’re already aware of this, but I just wanted to mention it. Since there has been some discussion about improving the error messages in general, this might be something to address in the future.

I added translations to common error messages from the backend, but especially the error messages originating from MongoDB and Fiber are not translated. Trying to also translate everything originating from the external libraries we are using would be cumbersome and probably not worth our time right now.

> Also, I observed that the language switches back to 'en' after refreshing the page, even if it was previously set to 'fi'. This could be another area to consider for future improvements.

Fixed.

Thank you for your useful and insightful comments!

amandahamynen reviewed (approved):
Fantastic! Everything works exactly as i hoped it would. Thanks for the great work, @VSirvio  . PR approved, and i’ll go ahead and merge it into development now.
